mir-ci/mir_ci/program.py

```diff
@@ -82,13 +83,18 @@ async def kill(self, timeout=default_term_timeout) -> None:
             pass
 
     async def __aenter__(self) -> 'Program':
+        def preexec_fn():
+            if self.preexec_fn:
+                self.preexec_fn()
+            os.setsid()
+
         process = await asyncio.create_subprocess_exec(
             *self.command,
             env=dict(os.environ, **self.env),
             stdout=asyncio.subprocess.PIPE,
             stderr=asyncio.subprocess.STDOUT,
             close_fds=True,
-            preexec_fn=os.setsid)
+            preexec_fn=preexec_fn)
```

Comment:
We still need this so that children get reaped:

```suggestion
            preexec_fn=preexec_fn,
            start_new_session=True)
```

Reply:
I am still doing `os.setsid()` in the preexec_fn, would the do the trick?

Reply:
This way you can let preexec_fn whol come from outside :)